urlb3290c8e9e_craigonic_jasl_tgz-pJ8-DriverJ8/src/jasl/ui/data/Player.java

```diff
@@ -0,0 +1,307 @@
+	 * @throws IllegalArgumentException in the case of an empty name or an
+	 * invalid (less than 1) entry turn argument.
+	 */
+
+	public Player(String name,Nationality.Nationalities nationality,
+	              int entryTurn)
+	{
+		// Copy the value of each remaining parameter to the
+		// corresponding variable if an exception is not found.
+
+		// Name
+
+		if (null == name)
+		{
+			throw new NullPointerException(nullPointerError);
+		}
+
+		if (0 == name.length())
+		{
+			throw new IllegalArgumentException(zeroLengthArgumentError);
+		}
+
+		_name = name;
+
+		// Nationality
+
+		_nationality = nationality;
+
+		// Entry turn
+
+		if (entryTurn < 1)
+		{
+			throw new IllegalArgumentException(invalidArgumentError +
+			                                   entryTurn);
+		}
+
+		_entryTurn = entryTurn;
+
+		_unitList = new ArrayList<String>();
+	}
+
+	// Public access methods
+
+	/**
+	 * Return a text representation of the attributes of a player.
+	 *
+	 * The output includes the labels associated with the units managed by
+	 * the player.
+	 *
+	 * @return a multi-line <CODE>String</CODE> specifying the current settings of an
+	 * instance of this type.
+	 *
+	 * @see #unitList
+	 */
+
+	public String toText()
+	{
+		StringBuffer returnString = new StringBuffer();
+
+		returnString.append("Player:\t" + toString() + "\n\n");
+		returnString.append("\tNationality:\t" + nationality() + "\n");
+		returnString.append("\tEntry Turn :\t" +
+		                     Integer.toString(_entryTurn) + "\n");
+
+		returnString.append("\n\tUnits:\n");
+
+		for (String unit : unitList())
+		{
+			returnString.append("\t\t" + unit + "\n");
+		}
+
+		// Return the completed string to calling program.
+
+		return returnString.toString();
+	}
+
+	/**
+	 * Return the name of a player.
+	 *
+	 * @return a <CODE>String</CODE> specifying the player's name.
+	 */
+
+	public String toString()
+	{
+		return _name;
+	}
+
+	/**
+	 * Return the nationality of the majority of the units that a player
+	 * controls.
+	 *
+	 * @return a <CODE>String</CODE> specifying the nationality.
+	 */
+
+	public String nationality()
+	{
+		return _nationality.toString();
+	}
+
+	/**
+	 * Return the turn that the units initially managed by a player will
+	 * enter the game.
+	 *
+	 * @return an <CODE>int</CODE> specifying the entry turn.
+	 */
+
+	public int entryTurn()
+	{
+		return _entryTurn;
+	}
+
+	/**
+	 * Return one of the units managed by a player.
+	 *
+	 * @param index the (zero-based) index of the <A HREF="../../counters/Unit.html">Unit</A> to be accessed
+	 *
+	 * @return the <CODE>Unit</CODE> at the specified index.
+	 *
+	 * @throws IndexOutOfBoundsException in the case where index is less than
+	 * zero or greater than the number of units in the list.
+	 *
+	 * @see #unitList
+	 */
+
+	public String unit(int index)
+	{
+		String unitToBeReturned;
+
+		try
+		{
+			unitToBeReturned = _unitList.get(index);
+		}
+
+		catch (IndexOutOfBoundsException exception)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)